darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -0,0 +1,358 @@
+    sns_plot.set_yticklabels(sns_plot.get_yticks(), rotation=0, size=6.5)
+    sns_plot.set_xlabel('Rank', size=8)
+    sns_plot.set_ylabel('Time (seconds)', size=8)
+    #sns.despine();
+    pyplot.savefig(buf, format='png', bbox_inches='tight')
+    buf.seek(0)
+    encoded.append(base64.b64encode(buf.read()))
+    return encoded
+  
+def main(args=None):
+
+    if args is None:
+        parser = argparse.ArgumentParser(description='')
+        setup_parser(parser)
+        args = parser.parse_args()
+
+    if args.debug:
+        print(args)
+
+    variables = {}
+    report = darshan.DarshanReport(args.input, read_all=True)
+    report.info()
+
+    #
+    # Setup template header variabels
+    #
+    variables['exe'] = report.metadata['exe']
+    variables['date'] = datetime.datetime.fromtimestamp(report.metadata['job']['start_time'], pytz.utc)
+    variables['jid'] = report.metadata['job']['jobid']
+    variables['uid'] = report.metadata['job']['uid']
+    variables['nprocs'] = report.metadata['job']['nprocs']
+    etime = int(report.metadata['job']['end_time'])
+    stime = int(report.metadata['job']['start_time'])
+    if etime > stime:
+        variables['runtime'] = etime - stime + 1
+    else:
+        variables['runtime'] = 0
+
+    if 'POSIX' in report.modules:
+        posix_df = report.records['POSIX'].to_df()
+    else:
+        posix_df = None
+
+    if 'MPI-IO' in report.modules:
+        mpiio_df = report.records['MPI-IO'].to_df()
+    else:
+        mpiio_df = None
+    
+    if 'STDIO' in report.modules:
+        stdio_df = report.records['STDIO'].to_df()
+    else:
+        stdio_df = None
+    
+    if 'APXC' in report.modules:
+        apxc_dict = report.records['APXC'].to_dict()
+    else:
+        apxc_dict = None
+
+    if 'APMPI' in report.modules:
+        apmpi_dict = report.records['APMPI'].to_dict()
+    else:
+        apmpi_dict = None
+
+    #
+    # Plot I/O cost
+    #
+    variables['plot_io_cost'] = plot_io_cost(posix_df, mpiio_df, stdio_df, int(variables['runtime']), int(variables['nprocs'])).decode('utf-8')
+
+    #
+    # Plot I/O counts
+    #
+    variables['plot_op_count'] = plot_op_count(posix_df, mpiio_df, stdio_df).decode('utf-8')
+
+    variables['fs_data'] = data_transfer_filesystem(report, posix_df, stdio_df)
+    apmpi_encoded = apmpi_process(apmpi_dict)
+    variables['apmpi_call_time'] = apmpi_encoded[0].decode('utf-8')
+    variables['apmpi_rank_totaltime'] = apmpi_encoded[1].decode('utf-8')
+
+    template_path = pkg_resources.path(darshan.templates, '')
+    with template_path as path:
+       loader = genshi.template.TemplateLoader(str(path))
+       template = loader.load('summary.html')
+
+       stream = template.generate(title='Darshan Job Summary', var=variables)
+       with open(args.output, 'w') as f:
+           f.write(stream.render('html'))
+           f.close()
+    return
+
+if __name__ == "__main__":
+    main()
```

Comment:
when invoked with the absolute path to a log file like so:

`python -m darshan.cli summary --output=report.html /home/tyler/LANL/rough_work/darshan/treddy_mpi-io-test_id545458_6-3-43062-1247823192538596076_13.darshan`

I get an error:

```
usage: darshan <command>
__main__.py: error: unrecognized arguments: /home/tyler/LANL/rough_work/darshan/treddy_mpi-io-test_id545458_6-3-43062-1247823192538596076_13.darshan
```

My call signature seems to match the one in the PR description though: `python -m darshan.cli summary --output=report.html log.darshan`

Looks like the PR description may need to be adjusted to at least include `--input` like this:

`python -m darshan.cli summary --output=report.html --input /home/tyler/LANL/rough_work/darshan/treddy_mpi-io-test_id545458_6-3-43062-1247823192538596076_13.darshan`

Even if I do that, I get another error:

```python
Traceback (most recent call last):
  File "/home/tyler/github_projects/spack/opt/spack/linux-ubuntu18.04-skylake_avx512/gcc-7.4.0/python-3.7.4-p6llyclmnre4iur3olezociwacltbqpc/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/tyler/github_projects/spack/opt/spack/linux-ubuntu18.04-skylake_avx512/gcc-7.4.0/python-3.7.4-p6llyclmnre4iur3olezociwacltbqpc/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/tyler/github_projects/darshan/darshan-util/pydarshan/darshan/cli/__main__.py", line 3, in <module>
    darshan.cli.main()
  File "/home/tyler/github_projects/darshan/darshan-util/pydarshan/darshan/cli/__init__.py", line 160, in main
    mod.main(args)
  File "/home/tyler/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py", line 342, in main
    apmpi_encoded = apmpi_process(apmpi_dict)
  File "/home/tyler/github_projects/darshan/darshan-util/pydarshan/darshan/cli/summary.py", line 158, in apmpi_process
    header_rec = apmpi_dict[0]
TypeError: 'NoneType' object is not subscriptable
```

The code may need some more graceful handling of scenarios where some instrumentation modules simply are not available so that I can take a quick look at what happens with a minimal DXT/POSIX trace.